tests/unit/reaction/test_combinatorics.py

```diff
@@ -206,3 +208,23 @@ def test_generate_permutations(
     assert spin_permutations[1][2][1] == +1
     assert spin_permutations[2][0][1] == +1
     assert spin_permutations[3][0][1] == +1
+
+
+def test_perform_external_edge_identical_particle_combinatorics(
+    y_to_d0_d0bar_pi0_pi0,
+):
+    result: Result = y_to_d0_d0bar_pi0_pi0("helicity")
+    some_graph = next(iter(result.transitions))
+    outgoing_edge_ids = some_graph.topology.outgoing_edge_ids
+    final_states = {
+        i: some_graph.get_edge_props(i)[0].name for i in outgoing_edge_ids
+    }
+    assert final_states == {3: "D0", 4: "pi0", 5: "D~0", 6: "pi0"}
```

Comment:
Not bad as a first test. Things would get more interesting if there would be final states with two pairs of identical particles  (such as pi-pi-pi+pi+) or 3 or 4 identical particles (such as 3/4 x pi0).